spec/struct.dd

```diff
@@ -493,24 +513,70 @@ constructor will have their `.init` values upon destruction.)
             const S1 c1 = S1(1);
 
             // Constructed mutable object is not implicitly convertible to immutable.
-            // immutable i1 = S1(1);
+            immutable i1 = S1(1); // error
 
             // Mutable constructor cannot construct immutable object.
-            // auto x1 = immutable S1(1);
+            auto x1 = immutable S1(1); // error
+
+
+            // Immutable constructor creates immutable object.
+            immutable i2 = immutable S2(1);
 
             // Immutable constructor cannot construct mutable object.
-            // auto x2 = S2(1);
+            auto x2 = S2(1); // error
 
             // Constructed immutable object is not implicitly convertible to mutable.
-            // S2 m2 = immutable S2(1);
+            S2 m2 = immutable S2(1); // error
 
             // Constructed immutable object is implicitly convertible to const.
             const S2 c2 = immutable S2(1);
+        }
+        ---
+        )
 
-            // Immutable constructor creates immutable object.
-            immutable i2 = immutable S2(1);
+        $(P Constructors can be overloaded with different attributes.)
+        $(SPEC_RUNNABLE_EXAMPLE_COMPILE
+        ---
+        struct S
+        {
+            this(int);           // non-shared mutable constructor
+            this(int) shared;    // shared mutable constructor
+            this(int) immutable; // immutable constructor
         }
-        ------
+
+        S m = S(1);
+        shared s = shared S(2);
+        immutable i = immutable S(3);
+        ---
+        )
+
+        $(P If the constructor can create a unique object (i.e. if it is `pure`),
+        the object is implicitly convertible to any qualifiers.
+        )
+        $(SPEC_RUNNABLE_EXAMPLE_COMPILE
+        ---
+        struct S
+        {
+            this(int) pure;
+            // Based on the definition, this creates a mutable object. But the
+            // created object cannot contain any mutable global data.
+            // Therefore the created object is unique.
+
+            this(int[] arr) immutable pure;
+            // Based on the definition, this creates an immutable object. But
+            // the argument int[] never appears in the created object so it
+            // isn't implicitly convertible to immutable. Also, it cannot store
+            // any immutable global data.
+            // Therefore the created object is unique.
```

Comment:
I think this should be reconsidered : https://issues.dlang.org/show_bug.cgi?id=10368 . Consider : 

```d
immutable int i;

struct C
{
    int* p;

    this(int) immutable pure
    { p = &i; }

}
void main()
{
    ++*C(9).p; // changes `i`
}
```
One cannot reliably construct a mutable object with an `immutable pure` constructor.

Reply:
It's a great point, but I am trying to document what is, not what should be. This should be addressed when dealing with 10368.